ownCloud/Migration/MigrationViewController.swift

```diff
@@ -61,14 +62,14 @@ class MigrationViewController: UITableViewController, Themeable {
 	func applyThemeCollection(theme: Theme, collection: ThemeCollection, event: ThemeEvent) {
 		self.tableView.applyThemeCollection(collection)
 		self.tableView.separatorColor = self.tableView.backgroundColor
+		headerLabel.applyThemeCollection(collection, itemStyle: .welcomeMessage)
 	}
 
 	// MARK: - User Actions
 
 	@IBAction func finishMigration() {
-		self.dismiss(animated: true) {
-			self.migrationFinishedHandler?()
-		}
+		self.migrationFinishedHandler?()
+		self.dismiss(animated: true)
```

Comment:
The problem here is, that the code `self.migrationFinishedHandler` needs to be called before the view controller disappears. Because the underlaying view controller will show the `MigrationViewController` again, because of unfinished migration state.

Reply:
I see it now. Thanks for explanation!